### PR TITLE
cmc added

### PR DESCRIPTION
**Pull request by SupremeSingh**

### CODE REVIEW

File: src/external/coinMarketCap.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import CoinMarketCap from 'coinmarketcap-api';
+import 
+{   QuotePriceData, 
+    CoinMarketCapTokenInfoResponse } from '../models/coinMarketCap';
+import { Optional } from 'src/types/optional';
+
+async function getTokenPrice(apiKey: string, tokenName: string, denominationName: string = 'USD'): Promise<Optional<QuotePriceData>> {
+  let tokenInfo = await getTokenInfo(apiKey, tokenName);
+  return tokenInfo!.data[tokenName].quote[denominationName];
+}
```
Comment (engineersamuel) [inline src/external/coinMarketCap.ts:16]: Remove the ! and handle if undefined

File: src/external/coinMarketCap.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+async function getTokenInfo(apiKey: string, tokenName: string): Promise<Optional<CoinMarketCapTokenInfoResponse>> {
+    let cmcClient: any = new CoinMarketCap(apiKey);
```
Comment (engineersamuel) [inline src/external/coinMarketCap.ts]: Avoid use of `any` if at all possible.  Furthermore, remove this here and put it at the top of the file, and remove the apiKey parameter and use the Envs object to reference this

File: src/utils/env.ts [hunk 3]
```diff
@@ -103,6 +106,13 @@ export namespace EnvUtils {
       // return false;
     }
 
+    if (!EnvUtils.isDefined(EnvUtils.EnvName.CMC_API_KEY)) {
+      console.warn(
+        'It may be necessary to set the environment variable "CMC_API_KEY"\n\tex. export CHAIN_SRC=https://hostname.com/chains/chain_logo_[chainid].[ext]'
```
Comment (engineersamuel) [inline src/utils/env.ts]: the first part of the text is fine, the 2nd part only applies to CHAIN_SRC and does not apply here, remove it.

**Review by engineersamuel — CHANGES_REQUESTED** [review]

File: src/external/coinMarketCap.ts (hunk 1 above)
Comment (SupremeSingh) [reply src/external/coinMarketCap.ts:16]: Added if Not block

File: src/utils/env.ts (hunk 3 above)
Comment (SupremeSingh) [reply src/utils/env.ts]: Fixed

File: src/external/coinMarketCap.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import CoinMarketCap from 'coinmarketcap-api';
+import 
+{   QuotePriceData, 
+    CoinMarketCapTokenInfoResponse } from '../models/coinMarketCap';
+import { EnvUtils } from '../utils/env';
+import { Optional } from 'src/types/optional';
```
Comment (engineersamuel) [inline src/external/coinMarketCap.ts]: You'll need to do '../types/optional'

**Review by engineersamuel — CHANGES_REQUESTED** [review]

File: src/external/coinMarketCap.ts (hunk 4 above)
Comment (SupremeSingh) [reply src/external/coinMarketCap.ts]: zwill correct in the next push

**Review by engineersamuel — APPROVED** [review]

**Pull request merged**